app/src/main/java/com/portfolio/tasky/entry/fragments/LoginFragment.kt

```diff
@@ -33,24 +35,85 @@ class LoginFragment : Fragment(), FragmentInflater by FragmentInflaterImpl() {
             openRegistrationFragment()
         }
     }
+    private fun setFieldValidations(loginFragment: LoginFragment?) {
+        clearValidatorParams()
+
+        val etEmail = viewBinding.etEmail.subLayout.etInput
+        etEmail.addTextChangedListener(loginFragment?.let {
+            TaskyValidationWatcherImpl(viewBinding.etEmail, LOGIN,
+                it
+            )
+        })
+
+        val etPassword = viewBinding.etPassword.subLayout.etInput
+        etPassword.addTextChangedListener(loginFragment?.let {
+            TaskyValidationWatcherImpl(viewBinding.etPassword, LOGIN,
+                it
+            )
+        })
+    }
+
+    private fun removeFieldValidations(loginFragment: LoginFragment?) {
+        clearValidatorParams()
+
+        val etEmail = viewBinding.etEmail.subLayout.etInput
+        etEmail.removeTextChangedListener(loginFragment?.let {
+            TaskyValidationWatcherImpl(viewBinding.etEmail, LOGIN,
+                it
+            )
+        })
+
+        val etPassword = viewBinding.etPassword.subLayout.etInput
+        etPassword.removeTextChangedListener(loginFragment?.let {
+            TaskyValidationWatcherImpl(viewBinding.etPassword, LOGIN,
+                it
+            )
+        })
+    }
+
+    override fun fieldsValidated(valid: Boolean) {
+        viewBinding.btnLogin.isEnabled = valid
+        valid.let {
+            if (it) {
+                viewBinding.etEmail.clearFocus()
+                viewBinding.etPassword.clearFocus()
+                viewBinding.btnLogin.requestFocus()
+            }
+        }
```

Comment:
There's an issue with this method. It's something I noticed on the video you added (Great you added video, love it)

When the email is valid, and then the password textfield gets verified, it's removing the focus.

No need to remove the focus in this case, since the user might be writing a longer password

Reply:
Correct. Rectified it!